tripy/tripy/utils/utils.py

```diff
@@ -214,6 +214,102 @@ def get_shape(data):
     return shape
 
 
+def get_stride(shape: Sequence[int], stride: Optional[Sequence[int]] = None) -> Optional[Sequence[int]]:
+    """
+    Get the stride for a given shape or return the provided stride.
+
+    Args:
+        shape: The shape of the tensor.
+        stride: The stride to use if provided. Defaults to None.
+
+    Returns:
+        The provided stride if given, otherwise the calculated canonical stride.
+    """
+    if stride is not None:
+        return tuple(stride)
+    return get_canonical_stride(shape)
+
+
+def get_canonical_stride(shape: Sequence[int]) -> Optional[Sequence[int]]:
+    """
+    Calculate the canonical stride for a given shape.
+
+    Args:
+        shape: The shape of the tensor.
+
+    Returns:
+        The calculated canonical stride.
+
+    Note:
+        For dimensions with size 0 or 1, the stride can be arbitrary.
+        This function returns 1 for dimensions with size 1 or 0, but other values are also valid.
+    """
+    if shape is None:
+        return None
+
+    if not shape:
+        return ()
+
+    canonical_stride = [1] * len(shape)
+    cumulative_product = 1
+
+    for dim_index in range(len(shape) - 1, -1, -1):
+        is_first_zero_dim = shape[dim_index] == 0 and dim_index != len(shape) - 1
+
+        if is_first_zero_dim:
+            canonical_stride[dim_index] = 1
+        elif shape[dim_index] != 1:
+            canonical_stride[dim_index] = cumulative_product
+
+        cumulative_product *= 1 if is_first_zero_dim else shape[dim_index]
+
+    return tuple(canonical_stride)
+
+
+def are_strides_equivalent(shape: Sequence[int], stride: Sequence[int], expected_stride: Sequence[int]) -> bool:
+    """
+    Compare the calculated stride with the expected stride, allowing for arbitrary values
+    in dimensions of size 0 or 1.
+
+    Args:
+        shape: The shape of the tensor.
+        stride: The stride to check.
+        expected_stride: The expected stride to compare against.
+
+    Returns:
+        True if strides are equivalent, False otherwise.
+    """
+    if len(shape) != len(stride) or len(shape) != len(expected_stride):
+        return False
+
+    return all(s == e or dim in (0, 1) for s, e, dim in zip(stride, expected_stride, shape))
+
+
+def is_canonical_stride(shape: Sequence[int], stride: Sequence[int]) -> bool:
+    """
+    Check if the given stride is canonical for the given shape.
+
+    A canonical stride is one where each dimension's stride is equal to
+    the product of all subsequent dimensions' sizes, except for dimensions
+    of size 0 or 1, which can have any stride.
+
+    Args:
+        shape: The shape of the tensor.
+        stride: The stride to check.
+
+    Returns:
+        True if the stride is canonical, False otherwise.
+
+    Note:
+        Allows arbitrary strides for dimensions with size 0 or 1 to account for framework discrepancies.
```

Comment:
@parthchadha @pranavm-nvidia It took me a while to figure this out that the stride support in PyTorch and the corresponding DL pack tensor is [broken](https://discuss.pytorch.org/t/strides-are-corrupted-after-to-dlpack-from-dlpack/178255). 
TVM seems to relax the check for dimensions with size = 1 [here](https://github.com/apache/tvm/pull/14797).

Reply:
I think @akhilg-nv also ran into this issue a while back.

Reply:
Yeah, I think @akhilg-nv's PR in mlir-tensorrt to relax the condition was merged.